libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -82,7 +84,7 @@ public static void TrustServiceUrl(string serviceUrl)
         /// <param name="expirationTime">The expiration time after which this service url is not trusted anymore.</param>
         public static void TrustServiceUrl(string serviceUrl, DateTime expirationTime)
         {
-            lock (_trustedHostNamesSync)
+            lock (_trustedHostNames)
```

Comment:
Any reason why don't we change `_trustedHostNames` to `ConcurrentDictionary` like the others so we can avoid this explicit `lock`?